vm/systemSmartContracts/staking.go

```diff
@@ -1706,6 +1712,54 @@ func (s *stakingSC) cleanAdditionalQueue(args *vmcommon.ContractCallInput) vmcom
 	return vmcommon.Ok
 }
 
+func (s *stakingSC) changeOwnerAndRewardAddress(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !s.flagValidatorToDelegation.IsSet() {
+		return vmcommon.UserError
+	}
+	if !bytes.Equal(args.CallerAddr, s.stakeAccessAddr) {
+		s.eei.AddReturnMessage("change owner and reward address can be called by validator SC only")
+		return vmcommon.UserError
+	}
+	if args.CallValue.Cmp(zero) != 0 {
+		s.eei.AddReturnMessage("CallValue must be 0")
+		return vmcommon.UserError
+	}
+	if len(args.Arguments) < 2 {
+		s.eei.AddReturnMessage("number of arguments is 2 at minimum")
+		return vmcommon.UserError
+	}
+
+	newOwnerAddress := args.Arguments[0]
+	if !core.IsSmartContractAddress(newOwnerAddress) {
+		s.eei.AddReturnMessage("new address must be a smart contract address")
+		return vmcommon.UserError
+	}
+
+	for i := 1; i < len(args.Arguments); i++ {
+		blsKey := args.Arguments[i]
+		registrationData, err := s.getOrCreateRegisteredData(blsKey)
+		if err != nil {
+			s.eei.AddReturnMessage(err.Error())
+			return vmcommon.UserError
+		}
+
+		if len(registrationData.RewardAddress) == 0 {
+			s.eei.AddReturnMessage("cannot unStake a key that is not registered")
```

Comment:
cannot changeOwnerAndRewardAddress a key that is not registered *

Reply:
:+1:

Reply:
done